.github/workflows/cpp.yml

```diff
@@ -210,7 +210,6 @@ jobs:
       ARROW_WITH_SNAPPY: ON
       ARROW_WITH_ZLIB: ON
       ARROW_WITH_ZSTD: ON
-      GTest_SOURCE: BUNDLED
```

Comment:
Note for self: next time we add something like this we probably want to add a comment on why we are adding it so we understand we can remove it later on. I didn't find a good enough reason on why it was added.
I had to go to a comment:
https://github.com/apache/arrow/pull/14177#discussion_r975965152
and a commit to see where was this coming from:
https://github.com/apache/arrow/pull/14536/commits/1e33ad94351c10c45980fc4ae1e9fd820f4f8b43